File: XPile 3-17/Programs/test_bitwise.py
from dataclasses import dataclass, field

def __ul_sort(arr, n): arr[:n] = sorted(arr[:n])

class __ul_Exception(Exception):
    def __init__(self, type_name, message):
        self.type_name = type_name
        self.message = message
        super().__init__(message)


def main():
    a = 255
    b = 15
    r1 = (a & b)
    print(r1)
    r2 = (240 | 15)
    print(r2)
    r3 = (255 ^ 15)
    print(r3)
    r4 = (~0)
    print(r4)
    r5 = (1 << 4)
    print(r5)
    r6 = (256 >> 4)
    print(r6)
    return 0


if __name__ == "__main__":
    main()

File: XPile 3-17/Programs/test_bitwise.js

function main() {
    let a = 255;
    let b = 15;
    let r1 = (a & b);
    console.log(r1);
    let r2 = (240 | 15);
    console.log(r2);
    let r3 = (255 ^ 15);
    console.log(r3);
    let r4 = (~0);
    console.log(r4);
    let r5 = (1 << 4);
    console.log(r5);
    let r6 = (256 >> 4);
    console.log(r6);
    return 0;
}

function __ul_sort(arr, n) { const sub = arr.slice(0, n).sort((a, b) => a < b ? -1 : a > b ? 1 : 0); for (let i = 0; i < n; i++) arr[i] = sub[i]; }

main();
